sdk/python/test_data/test_data_config.yaml

```diff
@@ -177,6 +177,15 @@ pipelines:
     - module: if_elif_else_complex
       name: lucky_number_pipeline
       execute: false
+    - module: pythonic_artifact_with_single_return
+      name: make_language_model_pipeline
+      execute: false
```

Comment:
I feel like these examples other than the one with list of artifact should also run on KFP since there's no change to the compiled result comparing to samples using `Input[]/Output[]`. Can we give them a try?

Reply:
I agree they should work. Thanks for suggesting we test it -- I had not.

It looks like there's one issue with the way this is implemented currently: the `ExecutorInput` provides the output file path in a temporary directory, rather than with a path starting with `/minio/` as expected.

```
"outputFile": "/tmp/kfp_outputs/output_metadata.json"
```

Do you have a suggestion for the preferred way to obtain the local task root from the KFP BE's `ExecutorInput` message?

![image](https://github.com/kubeflow/pipelines/assets/55268212/e2777520-2a6a-4e0f-8662-a64eba57264c)

Reply:
I wonder if `/minio` is a must. See my other comment.
Besides, I think it could be much easier to test through presubmit if you change change the yaml config here.

Reply:
Can you describe what you mean by `/minio` is a must? Not sure if you mean a must to check in these changes (I'm inclined to think some form of KFP BE support is a must to merge...) or if it's a must for the path to include `/minio`.

I'm not sure that the discussion on the [other thread](https://github.com/kubeflow/pipelines/pull/9932#discussion_r1332296355) solves this, since `dsl.get_uri('/tmp/kfp_outputs') == '/tmp/kfp_outputs'`, but possibly I'm interpreting the suggestion in the other thread incorrectly. In either case, I added (temporary) problematic test cases in [072ba3a](https://github.com/kubeflow/pipelines/pull/9932/commits/072ba3a5b67d681e0991e21fd064232fb707774a) to demonstrate the current undesirable behavior.

I added the pipelines as presubmit execution tests.

Reply:
I don't think using `/minio` as local path is a requirement to support remote url `minio://`. In v1, the artifact path has always been a temporary path like `/tmp/...`. This now reminds me that I had an related question on the design doc for this change, that is would backend be compatible/ready for the proposed SDK change. And looks like the conclusion was it doesn't needs backend changes to accommodate the SDK change?


> dsl.get_uri('/tmp/kfp_outputs') == '/tmp/kfp_outputs'

This might not be an issue? I think it might be okay that remote URL isn't a requirement or need to know by the SDK. 
I think we can verify it via the sdk execution test.
At this point, I see the following error from the last run:


```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.7/site-packages/kfp/dsl/executor_main.py", line 109, in <module>
    executor_main()
  File "/usr/local/lib/python3.7/site-packages/kfp/dsl/executor_main.py", line 101, in executor_main
    output_file = executor.execute()
  File "/usr/local/lib/python3.7/site-packages/kfp/dsl/executor.py", line 361, in execute
    result = self.func(**func_kwargs)
  File "/tmp/tmp.OKLmZtnT6W/ephemeral_component.py", line 9, in make_language_model
    import dill
ModuleNotFoundError: No module named 'dill'
I0921 02:31:42.399853      35 launcher_v2.go:151] publish success.
F0921 02:31:42.399894      35 main.go:49] failed to execute component: exit status 1
time="2023-09-21T02:31:43.290Z" level=info msg="sub-process exited" argo=true error="<nil>"
Error: exit status 1
time="2023-09-21T02:31:44.229Z" level=info msg="sub-process exited" argo=true error="<nil>"
Error: exit status 1
```

https://oss.gprow.dev/view/gs/oss-prow/pr-logs/pull/kubeflow_pipelines/9932/kubeflow-pipelines-sdk-execution-tests/1704683259281018880